editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -5743,47 +5757,69 @@ void CanvasItemEditorViewport::_perform_drop_data() {
 
 bool CanvasItemEditorViewport::can_drop_data(const Point2 &p_point, const Variant &p_data) const {
 	Dictionary d = p_data;
-	if (d.has("type")) {
-		if (String(d["type"]) == "files") {
-			Vector<String> files = d["files"];
-			bool can_instantiate = false;
-
-			List<String> scene_extensions;
-			ResourceLoader::get_recognized_extensions_for_type("PackedScene", &scene_extensions);
-			List<String> texture_extensions;
-			ResourceLoader::get_recognized_extensions_for_type("Texture2D", &texture_extensions);
-
-			for (int i = 0; i < files.size(); i++) {
-				String extension = files[i].get_extension().to_lower();
-
-				// Check if dragged files with texture or scene extension can be created at least once.
-				if (texture_extensions.find(extension) || scene_extensions.find(extension)) {
-					Ref<Resource> res = ResourceLoader::load(files[i]);
-					if (res.is_null()) {
-						continue;
-					}
-					Ref<PackedScene> scn = res;
-					if (scn.is_valid()) {
-						Node *instantiated_scene = scn->instantiate(PackedScene::GEN_EDIT_STATE_INSTANCE);
-						if (!instantiated_scene) {
-							continue;
-						}
-						memdelete(instantiated_scene);
-					}
+	if (d.has("type") && String(d["type"]) == "files") {
+		Vector<String> files = d["files"];
+		bool can_instantiate = false;
+		String drop_type;
+
+		List<String> scene_extensions;
+		ResourceLoader::get_recognized_extensions_for_type("PackedScene", &scene_extensions);
+		List<String> texture_extensions;
+		ResourceLoader::get_recognized_extensions_for_type("Texture2D", &texture_extensions);
+		List<String> audio_extensions;
+		ResourceLoader::get_recognized_extensions_for_type("AudioStream", &texture_extensions);
+
+		for (int i = 0; i < files.size(); i++) {
+			String extension = files[i].get_extension().to_lower();
+
+			// Optimization to not load a resource every time.
+			if (!(texture_extensions.find(extension) || scene_extensions.find(extension) || audio_extensions.find(extension))) {
+				continue;
+			}
+			// Check if dragged files with texture or scene extension can be created at least once.
+			Ref<Resource> res = ResourceLoader::load(files[i]);
+			if (res.is_null()) {
+				continue;
+			}
+			String type = res->get_class();
+			if (type == "PackedScene") {
+				Ref<PackedScene> sdata = Object::cast_to<PackedScene>(*res);
```

Comment:
```suggestion
				Ref<PackedScene> sdata = res;
```
More redundant casts. You can clean this up if you are changing the code anyway.